### PR TITLE
Add support for ARM, update dependencies

### PR DESCRIPTION
**Pull request by felixrieseberg**
This PR contains exactly what's in the title: I've added support for ARM on both Windows and macOS and updated a number of depenendencies.

I've also done some _very minor_ refactoring, most of which was required by `standard`.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -173,6 +186,7 @@ class Updates {
     const latest = {}
 
     const releases = await res.json()
+    console.log(releases)
```
Comment (malept) [inline index.js:189]: Should probably remove this before merging.
```suggestion
```

File: index.js [hunk 2]
```diff
@@ -219,10 +229,11 @@ class Updates {
       }
     }
 
-    return latest['darwin-x64'] || latest['win32-x64'] || latest['win32-ia32']
-      ? latest
-      : null
+    console.log(latest)
```
Comment (malept) [inline index.js:232]: More debug

```suggestion
```

File: index.js [hunk 3]
```diff
@@ -191,24 +206,19 @@ class Updates {
...
       }
-      if (latest['darwin-x64'] && latest['win32-x64'] && latest['win32-ia32']) {
+
+      if (hasAllAssets(latest)) {
         break
       }
     }
 
-    for (const key of ['win32-x64', 'win32-ia32']) {
+    for (const key of [PLATFORM_ARCH.WIN_X64, PLATFORM_ARCH.WIN_IA32]) {
```
Comment (malept) [inline index.js:219]: Should win-arm64 be included here?

File: index.js [hunk 4]
```diff
@@ -232,10 +243,36 @@ class Updates {
...
+}
+
+const hasAnyAsset = latest => {
+  return !!(
+    latest[PLATFORM_ARCH.DARWIN_X64] ||
+    latest[PLATFORM_ARCH.DARWIN_ARM] ||
+    latest[PLATFORM_ARCH.WIN_X64] ||
+    latest[PLATFORM_ARCH.WIN_IA32]
+  )
+}
```
Comment (malept) [inline index.js:262]: Should win-arm64 be added here?

**Review by malept — CHANGES_REQUESTED** [review]
I think `-arm` should be `-arm64` across the board, to be consistent with our usage of Node-style arch names.

File: index.js [hunk 5]
```diff
@@ -232,10 +240,36 @@ class Updates {
...
+  if (/.*(mac|darwin|osx).*(-arm).*\.zip/i.test(fileName)) {
+    return PLATFORM_ARCH.DARWIN_ARM64
+  }
+
+  if (/.*(mac|darwin|osx).*\.zip/i.test(fileName)) {
+    return PLATFORM_ARCH.DARWIN_X64
+  }
+
+  if (/win32-ia32/.test(fileName)) return PLATFORM_ARCH.WIN_IA32
+  if (/win32-arm/.test(fileName)) return PLATFORM_ARCH.WIN_ARM64
```
Comment (malept) [inline index.js:271]:
```suggestion
  if (/win32-arm64/.test(fileName)) return PLATFORM_ARCH.WIN_ARM64
```

File: index.js [hunk 6]
```diff
@@ -232,10 +240,36 @@ class Updates {
...
+  )
+}
+
+const hasAnyAsset = latest => {
+  return !!(
+    latest[PLATFORM_ARCH.DARWIN_X64] ||
+    latest[PLATFORM_ARCH.DARWIN_ARM64] ||
+    latest[PLATFORM_ARCH.WIN_X64] ||
+    latest[PLATFORM_ARCH.WIN_IA32]
+  )
```
Comment (malept) [inline index.js:258]:
```suggestion
  return !!(
    latest[PLATFORM_ARCH.DARWIN_X64] ||
    latest[PLATFORM_ARCH.DARWIN_ARM64] ||
    latest[PLATFORM_ARCH.WIN_X64] ||
    latest[PLATFORM_ARCH.WIN_IA32] ||
    latest[PLATFORM_ARCH.WIN_ARM64]
  )
```

File: index.js [hunk 7]
```diff
@@ -83,10 +97,9 @@ class Updates {
     const latest = await this.cachedGetLatest(account, repository, platform)
 
     if (!latest) {
-      const message =
-        platform === 'darwin-x64'
-          ? 'No updates found (needs asset matching *{mac,darwin,osx}*.zip in public repository)'
-          : 'No updates found (needs asset containing win32-{x64,ia32} or .exe in public repository)'
+      const message = platform.includes(PLATFORM.DARWIN)
+        ? 'No updates found (needs asset matching *{mac,darwin,osx}*.zip in public repository)'
+        : 'No updates found (needs asset containing win32-{x64,ia32,arm} or .exe in public repository)'
```
Comment (malept) [inline index.js:102]:
```suggestion
        : 'No updates found (needs asset containing win32-{x64,ia32,arm64} or .exe in public repository)'
```

**Review by malept — CHANGES_REQUESTED** [review]
I think these should be the last changes and then we should be good to go.

**Review by malept — APPROVED** [review]
In a follow-up PR, we should move to GitHub Actions.

:shipit:

**Comment by marcello3d** [comment]
Will this work with a macOS universal app? E.g. both arm + x64 in one .zip?

**Comment by felixrieseberg** [comment]
> Will this work with a macOS universal app? E.g. both arm + x64 in one .zip/.app?

Not right now, but a PR would be welcome!

**Comment by tazzben** [comment]
Does someone know when this will be deployed to the server?

**Pull request merged**